trie/verkle.go

```diff
@@ -457,8 +457,11 @@ func (t *VerkleTrie) UpdateContractCode(addr common.Address, codeHash common.Has
 
 		// Reuse the calculated key to also update the code size.
 		if i == 0 {
+			var basicDataKey [32]byte
+			copy(basicDataKey[:], key[:31])
+			basicDataKey[31] = utils.BasicDataLeafKey
 			// XXX add subfield update api
-			basicdata, err := t.root.Get(key, nil)
+			basicdata, err := t.root.Get(basicDataKey[:], nil)
```

Comment:
This was a bug. The original `key` last byte is `0x80` since that's the key of the current place of the account header where to store the first code-chunks.

The intended `basicdata` length check must pull the key with last byte `0x00`.

Reply:
oh that explains soooo much